webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -75,6 +108,24 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     [experimentFilters],
   );
 
+  const resetPagination = useCallback(() => {
+    setIsLoading(true);
+    setPage(0);
+    setExperiments([]);
+  }, []);
+
+  const onSortChange = useCallback(
+    (sorts: Sort[]) => {
+      setSorts(sorts);
+      const newSortString = makeSortString(sorts.filter(isValidSort));
+      setSortString(newSortString);
+      if (newSortString !== sortString) {
+        resetPagination();
+      }
```

Comment:
nit: I know that setState is async and so the if statement will probably execute before newSortString gets set, but it would still make more sense to switch those around imo